### ISSUE TITLE
Fetch metadata of stopped magnets

### ISSUE THREAD

**Issue by legacychimera247** [transmission/transmission#218]
azureus vuze has the ability to let me choose which files i want to download from a magnet file before starting the download...transmission just start the download of the entire file...would be nice to have that feature in transmission too...

an example : click on magnet , azureus opens and download magnet infos, ask what to download from magnet and then starts to download...transmission just start without asking...

**Comment by killemov** [comment]
So what you actually want when you add a magnet in paused mode is to have Transmission download the metadata and then stop. Why not ask for that?

**Comment by nishantkumarsah** [comment]
Why is this closed? What is the solution ?

**Comment by mikedld** [comment]
Looks pretty open to me @nishantkumarsah

**Comment by mirkorm** [comment]
Yes, I'd like this feature to be implemented too, it's very useful, especially when there are a lot of large files in the torrent and you just need some of them, and this is a very common scenario from what I know.

Regards

**Comment by elisimpson** [comment]
This would be super useful. What I have to do now is start the torrent from the magnet link, wait for it to download the metadata and inevitably part of the file data as well, then immediately stop the torrent, select the files I want, delete any unnecessary pieces that already downloaded, restart the torrent.

**Comment by pataquets** [comment]
@elisimpson : you might want to keep an eye on #852 related with what you currently do as workaround

**Comment by elisimpson** [comment]
@pataquets : I actually found a better workaround, which is start the magnet link with qbittorrent, which can download the metadata without starting the torrent. Then I save the metadata to a new torrent file and load it in transmission.

**Comment by DidierLoiseau** [comment]
Just adding my 2 cents: there is an [old ticket on trac](https://trac.transmissionbt.com/ticket/4808) for this as well, and it has an attached patch but it is 7 years old…

@elisimpson you can also use a service such as [Magnet >> Torrent][1] to do that.

  [1]: http://magnet2torrent.com/

**Comment by Raymo111** [comment]
Can we get this PR merged please? ^^

**Comment by rheicide** [comment]
Please correct me if I'm wrong but it looks like magnets are not having their metadata fetched after being added with "Start when added" unchecked, which in my understanding is the goal of this issue.

If this is the case, it's probably related to the fact that the `prefetchMagnetMetadata` flag introduced by #1080 was later removed by #3013.

**Comment by fmnm04** [comment]
> If this is the case, it's probably related to the fact that the `prefetchMagnetMetadata` flag introduced by #1080 was later removed by #3013.

Can someone give an explanation as to why this feature was removed? It's annoying to deal with having to download part of a torrent files just to see their metadata. And why is this issue closed if the problem hasn't been solved?

**Comment by GaryElshaw** [comment]
The issue is alive elsewhere. The function itself opened a lot of debate. Some users made the case that sometimes metadata is 'data-sized'; and others that zero data should be allowed if start when added is unchecked. I hope this PR is revisited and added #4874. CC @Coeur
